source/grid/grid_out.cc

```diff
@@ -2995,6 +3035,27 @@ namespace
 
 
 
+  /**
+   * Return all non-internal faces of a triangulation.
+   */
+  template <int dim, int spacedim>
+  std::vector<typename Triangulation<dim, spacedim>::active_face_iterator>
+  all_faces(const Triangulation<dim, spacedim> &tria)
```

Comment:
`get_faces()`. But on second thought, I think you want to qualify this: you're not returning all faces or edges, but in fact only the ones on the boundary.